build-aux/m4/ax_boost_system.m4

```diff
@@ -68,7 +68,6 @@ AC_DEFUN([AX_BOOST_SYSTEM],
 					   ax_cv_boost_system,
         [AC_LANG_PUSH([C++])
 			 CXXFLAGS_SAVE=$CXXFLAGS
-			 CXXFLAGS=
```

Comment:
isn't `CXXFLAGS` restored at line 75?

Reply:
Yes. But the important is to not have it cleared for compilation, as, otherwise, the compilation will fail with one of these `__float128`-is-undefined errors that we saw with `clang-3.8` and no further flags.

 I guess saving and restoring `CXXFLAGS` isn't even strictly necessary anymore, but interestingly enough, in upstream autoconf-archive, the commit that caused the regression also only added the `CXXFLAGS=` line, see here for reference:

http://git.savannah.nongnu.org/cgit/autoconf-archive.git/commit/m4?id=d48ec2210047bbf17702700811c3f16a5369e06c

So I assumed the best is to basically just revert that part of that commit.

Reply:
The changes to `build-aux/m4/ax_boost_system.m4` contained in this PR fix the configure problem described, i.e. mis-detection of libboost system during the configure step.